src/dotenv.ts

```diff
@@ -40,6 +40,10 @@ export type Env = typeof process.env;
  * If you need more control (or access to the values), consider using `loadDotenv` instead
  *
  */
+const dotenvVariableRegistry = new Map<
+  Record<string, any>,
+  Record<string, boolean>
```

Comment:
This could be a `Set<string>` ?

Also can we maybe make a util `getDotEnvVariableRegistry(env)` that stores it in globalThis to use in both places? 🙏🏼